authenticate/authenticate.go

```diff
@@ -75,7 +75,7 @@ func New(opts config.Options) (*Authenticate, error) {
 	if err != nil {
 		return nil, err
 	}
-	redirectURL := opts.AuthenticateURL
+	redirectURL := *opts.AuthenticateURL
```

Comment:
Maybe add a warning comment here if we keep this as-is.

Reply:
> The deref in Authenticate fixes the immediate problem, but I'm inclined to pass in a fully dereferenced Options struct to the services. Once inside, we can deal in pointers if needed.

> That said, this looks fine, but I'd favor the value based approach as a safer pattern. Thoughts?

I agree. Let me take another pass.